### ISSUE TITLE
CustomDump или нужно больше данных

### ISSUE THREAD

**Issue by KrapivinAndrey** [Pr-Mex/vanessa-automation#1473]
```Gherkin
Функционал: Больше данных для расследования падений сценария
   Как разработчик
   Я хочу получать больше данных если возникают ошибки

Сценарий: Произвольные данные при падении
  Допустим Есть сценарий
  Когда Падает какой-то шаг
  Тогда для отчета аллюр создается куча полезных файлов: сетевые соединения, описание форм, значения переменных и тд
  И **мой произвольный дамп** каких-либо специфичных настроек

```

Решаемая задача:

возможно нужны какие-то дополнительные сведения о системе, состояние настроек и тд.

Варианты решения:

1. Вызов серверной функции, которая формирует табдок - указание этой функции в настройках.
2. Вызов внешней обработки. Надо быть аккуратным на серверных базах.

**Comment by Pr-Mex** [comment]
@KrapivinAndrey 
Я думаю надо подключать внешнюю обработку, которая будет формировать нужные данные.
Получается:
1. Надо создать настройку VA, например, "ВнешнийОбработчикОшибки", где указывается путь к обработке.
2. Надо, чтобы обработка имела экспортный метод ПриВозникновенииОшибки(Ванесса, ДопПараметры). VA будет вызывать этот метод, когда упал шаг сценария. В ДопПараметры надо передать всё, что нужно для регистрации ошибки.
3. Чтобы работало в серверных базах надо включить обработку в подключение в том же месте, где подключаются плагины.

**Comment by KrapivinAndrey** [comment]
@Pr-Mex 

2. ПриВозникновенииОшибки должна будет возвращать ТабДок для прикладывания к отчету аллюр или просто как артефакт сборки.
3. А если это Single? я бы добавил оба варианта на всякий случай.

**Comment by Pr-Mex** [comment]
> @Pr-Mex
> 
> 2. ПриВозникновенииОшибки должна будет возвращать ТабДок для прикладывания к отчету аллюр или просто как артефакт сборки.

@KrapivinAndrey 
Думаю ПриВозникновенииОшибки() должна как раз и выполнить нужные действия.
Она может вызвать Ванесса.ПриложитьМакетКAllure() (уже есть что-то такое)
или
Ванесса.Что-тоДругое()
и т.д.
Или просто приложить к сборке файлы самостоятельно не обращаясь к Ванесса.

**Comment by Pr-Mex** [comment]
> 3. А если это Single? я бы добавил оба варианта на всякий случай.

@KrapivinAndrey 
Да, Single не выполняет подключение плагинов, т.к. они уже встроены в него.
Поэтому этот момент надо будет отдельно учитывать. Это могу взять на себя.

**Comment by Pr-Mex** [comment]
Не актуально
